### ISSUE TITLE
ci: the native dylib end-to-end tests never run in CI

### ISSUE THREAD

**Issue by csm** [csm/clojurust#403]
Raised in review of #402.

`crates/cljrs/tests/pinned_dylib_e2e.rs` holds the end-to-end tests for `:rust/load :dylib` (pinned `@<sha>` resolution, plain `require`, and the `:local/root` working-tree cases, including #402's `local_root_native_dep_picks_up_an_edit_that_keeps_its_mtime`). Each test returns early unless `CLJRS_DYLIB_E2E=1` is set, and no workflow sets it:

- `ci.yml`, `stress-test.yml` and `nightly.yml` all run plain `cargo test --workspace`, so all seven tests pass as no-ops.
- `nightly.yml`'s `ci` job only runs when `should_release == 'true'`, so it wouldn't be a reliable place to run them either: it is skipped on days with only docs/www changes.

The result is that nothing guards `native::pinned` against regressions on real cargo builds; the unit tests cover the pure pieces (planning, digest, record logic) but not cargo's behaviour.

### Cost

Measured locally: ~4 minutes for the whole file, ~50s for a single `:local/root` test. Most of it is each test compiling `cljrs-interop` and its dependency tree into that test's private `HOME` cache.

### Options

1. **Scheduled workflow** (plus `workflow_dispatch`), running daily on `main`:
   ```sh
   CLJRS_DYLIB_E2E=1 cargo test -p cljrs --test pinned_dylib_e2e
   ```
   No PR latency; a regression is caught within a day rather than blocking the PR that caused it.
2. **Path-filtered PR job** in `ci.yml`, running the same command only when `crates/cljrs/src/native/**`, `crates/cljrs-interop/**`, `crates/cljrs-project/src/vcs*` or `crates/cljrs/tests/pinned_dylib_e2e.rs` change. Catches the regression on the PR that introduces it, at the cost of ~4 minutes on those PRs.

Both can coexist (1 catches breakage from changes outside the filtered paths, e.g. `cljrs-interop`'s transitive deps).

### Notes

- The tests need `git` and a Rust toolchain on the runner, both already present on `ubuntu-latest` with `dtolnay/rust-toolchain`.
- They set `HOME` per test, so `Swatinem/rust-cache` won't cover the wrapper builds; the time above is what to expect.
